modules/lang-painless/src/main/java/org/elasticsearch/painless/AnalyzerCaster.java

```diff
@@ -30,6 +30,9 @@
 public final class AnalyzerCaster {
 
     public static Cast getLegalCast(Location location, Type actual, Type expected, boolean explicit, boolean internal) {
+        if (actual == null || expected == null) {
```

Comment:
Do you have a test case that triggers this? because it's definitely worth investigating.  Since it's an unexpected case maybe Objects.requireNonNull makes more sense to keep the code a bit cleaner?  I could go either way since you're trying to add a better error message, though.